File: examples/src/Shapes/SetFileTypeForAnEmbeddingObject.cs
import aspose.slides as slides
using Aspose.slides.DOM.Ole
using Aspose.slides.Examples.CSharp
using Aspose.slides.Export
using System
using System.Collections.Generic
using System.IO
using System.Linq
using System.text
using System.Threading.Tasks

namespace CSharp.shapes
{
    class SetFileTypeForAnEmbeddingObject
    {
        public static void Run() {

            #ExStart:SetFileTypeForAnEmbeddingObject

            with slides.Presentation() as pres:
            {
                # The path to the documents directory.
                dataDir = RunExamples.GetDataDir_Shapes()

                # Add known Ole objects
                byte[] fileBytes = File.ReadAllBytes(dataDir + "test.zip")

                # Create Ole embedded file info
                IOleEmbeddedDataInfo dataInfo = new OleEmbeddedDataInfo(fileBytes, "zip")

                # Create OLE object
                IOleObjectFrame oleFrame = pres.slides[0].shapes.AddOleObjectFrame(150, 20, 50, 50, dataInfo)
                oleFrame.IsObjectIcon = True


                pres.save(dataDir + "SetFileTypeForAnEmbeddingObject.pptx", slides.export.SaveFormat.PPTX)
            }

            #ExEnd:SetFileTypeForAnEmbeddingObject

        }
    }
}

File: examples/src/Shapes/set_file_type_for_an_embedding_object.py
import aspose.slides as slides


def set_file_type_for_an_embedding_object(global_opts):
    with slides.Presentation() as pres:
        # Add known Ole objects
        with open(global_opts.data_dir + "test.zip", "rb") as file:
            file_bytes = file.read()

            # Create Ole embedded file info
            data_info = slides.dom.ole.OleEmbeddedDataInfo(file_bytes, "zip")

            # Create OLE object
            ole_frame = pres.slides[0].shapes.add_ole_object_frame(150, 20, 50, 50, data_info)
            ole_frame.is_object_icon = True

        pres.save(global_opts.out_dir + "shapes_set_ole_object_out.pptx", slides.export.SaveFormat.PPTX)
